pytorch_lightning/callbacks/pruning.py

```diff
@@ -0,0 +1,282 @@
+    import torch.nn.utils.prune as pytorch_prune
+
+
+_PYTORCH_PRUNING_FUNCTIONS = {
+    "ln_structured": pytorch_prune.ln_structured,
+    "l1_unstructured": pytorch_prune.l1_unstructured,
+    "random_structured": pytorch_prune.random_structured,
+    "random_unstructured": pytorch_prune.random_unstructured,
+}
+
+_PYTORCH_PRUNING_METHOD = {
+    "ln_structured": pytorch_prune.LnStructured,
+    "l1_unstructured": pytorch_prune.L1Unstructured,
+    "random_structured": pytorch_prune.RandomStructured,
+    "random_unstructured": pytorch_prune.RandomUnstructured,
+}
+
+
+def check_parameters_to_prune(
+    pl_module: LightningModule,
+    parameters_to_prune: Optional[List],
+    parameters: List[str] = ["weight"]
+) -> List:
+
+    is_parameters_to_prune_none = parameters_to_prune is None
+    current_modules = [m for m in pl_module.modules() if not isinstance(m, (LightningModule, ModuleDict, ModuleList))]
+
+    if parameters_to_prune is None:
+        parameters_to_prune = []
+        for p in parameters:
+            for m in current_modules:
+                param = getattr(m, p, None)
+                if param is not None:
+                    parameters_to_prune.append((m, p))
+
+    if isinstance(parameters_to_prune, list) and not is_parameters_to_prune_none:
+
+        if all(isinstance(p, (list, tuple)) and len(p) == 2 for p in parameters_to_prune):
+
+            missing_modules = []
+            missing_parameters = []
+
+            for module, param_name in parameters_to_prune:
+                if module not in current_modules:
+                    missing_modules.append(module)
+                    continue
+
+                parameter = getattr(module, param_name)
+
+                if parameter is None:
+                    missing_parameters.append(parameter)
+
+            if len(missing_modules) > 0 or len(missing_parameters) > 0:
+                raise MisconfigurationException(
+                    "Ths provided parameters_to_tune doesn't exist in the model. "
+                    f"Found mismatching modules: {missing_modules} and missing_parameters: {missing_parameters}"
+                )
+
+    return parameters_to_prune
+
+
+class PruningCallback(Callback):
+
+    PARAMETER_NAMES = ["weight", "bias"]
+
+    def __init__(
+        self,
+        pruning_fn: Union[Callable, str],
+        parameters_to_prune: Optional[List[Union[str, Tuple[nn.Module, str]]]] = None,
+        parameter_names: List[str] = ["weight"],
+        use_global_unstructured: bool = True,
+        amount: Optional[Union[int, float]] = 0.5,
+        prune_on_epoch_end: Optional[bool] = False,
+        prune_on_fit_end: Optional[bool] = True,
+        make_pruning_permanent: Optional[bool] = True,
+        use_lottery_ticket_hypothesis: Optional[bool] = True,
+        pruning_dim: Optional[int] = None,
+        pruning_norm: Optional[int] = None,
+    ) -> None:
+        """
+
+        Pruning Callback relying on PyTorch prune utils.
```

Comment:
Insert link to https://pytorch.org/tutorials/intermediate/pruning_tutorial.html maybe?

Reply:
Done !